aopy/data.py

```diff
@@ -702,6 +703,112 @@ def load_electrode_pos(data_dir, pos_file):
     y_pos = electrode_pos['topdown_y'].to_numpy()
     return x_pos, y_pos
 
+def map_acq2elec(signalpath_table, acq_ch_subset=None):
+    '''
+    Create index mapping from acquisition channel to electrode number. 
+    Excel files can be loaded as a pandas dataframe using pd.read_excel
+    
+    Args:
+        signalpath_table (pd dataframe): Signal path information in a pandas dataframe. (Mapping between electrode and acquisition ch)
+        acq_ch_subset (nacq): Subset of acquisition channels to call. If not called, all acquisition channels and connected electrodes will be return. If a requested acquisition channel isn't returned a warned will be displayed
+
+    Returns:
+        acq_chs (nelec): Acquisition channels that map to electrodes (e.g. 240/256 for viventi ECoG array)
+        connected_elecs (nelec): Electrodes used (e.g. 240/244 for viventi ECoG array)   
+    '''    
+    # Parse acquisition channels used and the connected electrodes
+    connected_elecs_mask = np.logical_not(np.isnan(signalpath_table['acq']))
+    connected_elecs = signalpath_table['electrode'][connected_elecs_mask].to_numpy()
+    acq_chs = signalpath_table['acq'][connected_elecs_mask].to_numpy(dtype=int)
+
+    if acq_ch_subset is not None:
+        acq_chs_mask = np.where(np.in1d(acq_chs, acq_ch_subset))[0]
+        acq_chs = acq_chs[acq_chs_mask]
+        connected_elecs = connected_elecs[acq_chs_mask]
+        if len(acq_chs) < len(acq_ch_subset):
+            missing_acq_chs = acq_ch_subset[np.in1d(acq_ch_subset,acq_chs, invert=True)]      
+            warning_str = "Requested acquisition channels " + str(missing_acq_chs) + " are not connected"
+            warnings.warn(warning_str)
```

Comment:
That was the impetus for including it. It doesn't break/stop the function, it just displays a warning that some requested channels (and which ones) aren't connected since the user can't know beforehand.